AI/Utilities/RemovalLogic.py

```
@@ -342,6 +342,112 @@ def _creature_keywords(creature: dict) -> set[str]:
+    """Ward on this card, or None if it has none.
+
+    Returns `{"mana": int | None, "text": str}` where `mana` is the mana needed
+    to pay the ward, or None for a ward whose cost we cannot pay through the UI
+    (pay life, discard a card, ...). `{"mana": None}` therefore means "warded and
+    we must not target it", not "free".
+    """
+    if grp_id is None:
+        return None
+    try:
+        info = CardInfo.get_card_info(grp_id) or {}
+    except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)